src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/design/hideutilityclassconstructor/Example2.java

```diff
@@ -0,0 +1,47 @@
+/*xml
+<module name="Checker">
+  <module name="TreeWalker">
+    <module name="HideUtilityClassConstructor">
+      <property name="ignoreAnnotatedBy"
+        value="SpringBootApplication, java.lang.Deprecated" />
```

Comment:
> Annotation names provided in this property must exactly match the annotation names on the classes. If the target class has annotations specified with their fully qualified names (including package), the annotations in this property should also be specified with their fully qualified names. Similarly, if the target class has annotations specified with their simple names, this property should contain the annotations with the same simple names.

Show some of these issues in your example.

I assume `Deprecated` by itself won't work. Show that and put a comment to that effect.

Reply:
Can you also provide me where we said we would be so strict?

Reply:
> I assume Deprecated by itself won't work.

Only `Deprecated` will not work if it doesn't match the annotation used.

Reply:
> Can you also provide me where we said we would be so strict?

We mentioned this at https://github.com/checkstyle/checkstyle/issues/14424#issuecomment-2008610998. We agreed to do exact match first as https://github.com/checkstyle/checkstyle/pull/14553. We might do pattern matching later, but they need to solve the problem of https://github.com/checkstyle/checkstyle/issues/5434 first.

Reply:
> > Annotation names provided in this property must exactly match the annotation names on the classes. If the target class has annotations specified with their fully qualified names (including package), the annotations in this property should also be specified with their fully qualified names. Similarly, if the target class has annotations specified with their simple names, this property should contain the annotations with the same simple names.
> 
> Show some of these issues in your example.
> 
> I assume `Deprecated` by itself won't work. Show that and put a comment to that effect.

done

Reply:
This does not look done. I don't see any examples were we fail to match because partials was only provided.

It is also not documented that we understand `java.lang` when we say `property must exactly match`. `Deprecated` is not an exact match for `java.lang.Deprecated`.

Reply:
I removed my comment here and added some more below.